### ISSUE TITLE
Composer dependencies

### ISSUE THREAD

**Issue by stevector** [paulgibbs/behat-wordpress-extension#66]
I maintain some of Pantheon's WordPress plugins, particularly the Behat tests included in them. I also have a lot of experience with Drupal Behat extension that inspired the architecture here. So I'm very excited to try incorporating this repo into our test suites. Right now I'm slightly blocked on Composer version constraints.

## Expected behavior

From a checkout of [solr-power](https://github.com/pantheon-systems/solr-power), I'd like to be able to run `composer require paulgibbs/behat-wordpress-extension` and have the command complete. 

## Current behavior

Right now I'm getting the following error:

```
  Problem 1
    - Installation request for paulgibbs/behat-wordpress-extension ^0.5.0 -> satisfiable by paulgibbs/behat-wordpress-extension[v0.5.0].
    - Conclusion: remove behat/behat 3.2.x-dev
    - Conclusion: don't install behat/behat 3.2.x-dev
    - paulgibbs/behat-wordpress-extension v0.5.0 requires behat/behat ~3.1.0 -> satisfiable by behat/behat[v3.1.0, v3.1.0rc1, v3.1.0rc2].
    - Can only install one of: behat/behat[v3.1.0, 3.2.x-dev].
    - Can only install one of: behat/behat[v3.1.0rc1, 3.2.x-dev].
    - Can only install one of: behat/behat[v3.1.0rc2, 3.2.x-dev].
    - Installation request for behat/behat (locked at 3.2.x-dev) -> satisfiable by behat/behat[3.2.x-dev].
```

The conflict is happening because our suite of Behat tests ([pantheon-systems/pantheon-wordpress-upstream-tests](https://github.com/pantheon-systems/pantheon-wordpress-upstream-tests/blob/master/composer.json)) is set to `"behat/behat": "^3.1"` which allows for updates to newer versions of Behat like `3.2.0` or the latest `3.3.0`. This repository uses the more restrictive `~3.1.0`

## Possible solution

I could force solr-power to go back to `3.1.0` but I'd rather clarify if the requirements here intentionally set to stay on `3.1.0`. If `^3.1.0` is acceptable, [I have a branch ready for a PR](https://github.com/stevector/behat-wordpress-extension/tree/composer-reqs).

## Steps to reproduce (for bugs)
1. Clone https://github.com/pantheon-systems/solr-power
2. `composer install`
3. `composer require paulgibbs/behat-wordpress-extension`

You will see the error message pasted above.

I've tested my suggested fix with:

```
composer config repositories.wp-behat  vcs    git@github.com:stevector/behat-wordpress-extension.git
composer require paulgibbs/behat-wordpress-extension:dev-composer-reqs
```

## Context

Pantheon has [a shared suite of Behat tests](https://github.com/pantheon-systems/pantheon-wordpress-upstream-tests) that we use within our plugins. I've never been satisfied with it, largely because I've been comparing it to the Drupal Behat Extension. I would like to consider refactoring our WordPress Behat tests around this repository and contribute improvements where necessary.

## Your environment

I work on a Mac but our WordPress plugins have tests that run in CircleCI and Travis.

[pantheon-wordpress-upstream-tests](https://github.com/pantheon-systems/pantheon-wordpress-upstream-tests) is the repo of shared tests and context that are reused in the following: 

Plugins:

- https://github.com/pantheon-systems/wp-saml-auth
- https://github.com/pantheon-systems/solr-power
- https://github.com/pantheon-systems/wp-native-php-sessions
- https://github.com/pantheon-systems/wp-redis
- https://github.com/lcache/wp-lcache

Copies of core:

- https://github.com/pantheon-systems/pantheon-wordpress-develop
- https://github.com/pantheon-systems/WordPress
- https://github.com/pantheon-systems/wordpress-network

**Comment by paulgibbs** [comment]
Hi @stevector

Thanks for reaching out! I've read a lot of the blog posts etc that Pantheon has written around Behat, so it's nice to have some contact. Given that, I'd been thinking about trying to find someone at Pantheon to talk to about WordHat, so this is very convenient. :)

The Behat version requires was more set to avoid upstream breaking changes. I'll have a look at what's new in Behat this evening/weekend, and maybe chat with co-conspirator @stephenharris, but I would guess it's pretty likely that we'd be up for changing it.

Any help building WordHat itself would be very much appreciated and something we're interested in, so let us help you with this first, and I guess we'll see where things go! We have a Slack set-up should you feel particularly WordHatty: http://wordhat.herokuapp.com :)

**Comment by stevector** [comment]
Thanks @paulgibbs! I'll open the PR for the dependency change.

I found WordHat because @ataylorme asked me to take a look at https://github.com/xwp/wp-js-widgets/pull/43 (which I see you've found too). I still need to read that PR in greater detail. I'm especially glad to have come across the repo now because I'm doing a Behat presentation at DrupalCon next month: https://events.drupal.org/baltimore2017/sessions/bending-behats-benefits

I've joined the Slack team and would be happy to chat further.

**Comment by stevector** [comment]
PR opened here: https://github.com/paulgibbs/behat-wordpress-extension/pull/67

**Comment by paulgibbs** [comment]
There aren't any Behat changes that require us to do work to support the newer versions, so I'm okay to change this. The only thing that will be an issue when Behat does a 4.x.x release is the removal of the `SnippetAcceptingContext` interface, which has been deprecated in 3.2.x. We currently support this, mainly as a convenience for people using WordHat, so we'll just have to remove this once Behat 4.x.x hits, which ought to be trivial.

**Comment by paulgibbs** [comment]
Merged! Thanks for the PR!
